velox/functions/sparksql/DecimalArithmetic.cpp

```diff
@@ -539,19 +577,46 @@ void registerDecimalAddSubtract(const std::string& name) {
       "max({a_scale}, {b_scale})",
       fmt::arg("a_scale", S1::name()),
       fmt::arg("b_scale", S2::name()));
-  registerDecimalBinary<Func>(name, makeConstraints(rPrecision, rScale));
+  return {rPrecision, rScale};
 }
-
 } // namespace
 
+template <typename TExec>
+using AddFunctionAllowPrecisionLoss = DecimalAddFunction<TExec, true>;
+
+template <typename TExec>
+using AddFunctionDenyPrecisionLoss = DecimalAddFunction<TExec, false>;
```

Comment:
The AddFunctionAllowPrecisionLoss and AddFunctionAllowPrecisionLoss are used only once. Do we need them?

Reply:
Yes, because it is template function, otherwise, we need to use previous ParameterBinder.